Automattic/pocket-casts-ios
# Tracks: Podcast settings events

| 📘 Project: #154 |
|:---:|

I introduced a debouncer class to make it easier to delay saving the Skip First / Last events until the user is done interacting with the setting.

### Events added
- `notifications_opt_in_shown`: When the system push notification alert is shown the user
- `notifications_opt_in_allowed`: When the user taps the allow button to allow push notifications
- `notifications_opt_in_denied`: When the user taps the don't allow button to reject push notifications
- `podcast_settings_feed_error_tapped`: When the user taps the feed error item in the podcast settings
- `podcast_settings_feed_error_update_tapped`: When the user taps the update button on the feed error alert
- `podcast_settings_feed_error_fix_succeeded`: If the feed error issue was fixed
- `podcast_settings_feed_error_fix_failed`: If the feed error issue failed for any reason
- `podcast_settings_auto_download_toggled`: When the user toggles the Auto Download option
- `podcast_settings_notifications_toggled`: When the user toggles the Notifications option
- `podcast_settings_auto_add_up_next_toggled`: When the user toggles the Add to Up Next option
- `podcast_settings_auto_add_up_next_position_option_changed`: When the user changes the Position in Queue setting
- `podcast_settings_custom_playback_effects_toggled`: When the user toggles the custom playback effects
- `podcast_settings_skip_first_changed`: When the user changes the skip first option
- `podcast_settings_skip_last_changed`: When the user changes the skip last option
- `podcast_settings_auto_archive_toggled`: When the user toggles the custom Auto Archive option
- `podcast_settings_auto_archive_played_changed`: When the user changes the Archive Played Episodes option
- `podcast_settings_auto_archive_inactive_changed`: When the user changes the Archive Inactive Episodes option
- `podcast_settings_auto_archive_episode_limit_changed`: When the user changes the Archive episode limit option
- `podcast_settings_siri_shortcut_added`: When the user adds a siri shortcut from the podcast settings
- `podcast_settings_siri_shortcut_removed`: When the user removes a siri shortcut from the podcast settings

## To test

### Notification opt in / out
Note: If you have enabled notifications before you will need to uninstall and reinstall the app

1. Launch the app
3. Tap on the Podcasts tab
4. Tap on any podcast
5. Tap the Cog to open the settings
6. Tap the Notifications item
7. ✅ `🔵 Tracked: notifications_opt_in_shown`
8. Tap Allow
9. ✅ `🔵 Tracked: notifications_opt_in_allowed`
10. Uninstall and repeat the steps instead tapping the Don't Allow button
11. ✅ `🔵 Tracked: notifications_opt_in_denied`

### Settings changed
#### Setup
1. Open Xcode
2. Open `PodcastSettingsViewController+Table.swift` and go to line 392
3. Comment out line 392 and 394
4. This will force the feed error to show always

#### Testing
1. Launch the app
6. Tap on the Podcasts tab
7. Tap on any podcast
8. Tap the Cog to open the settings
9. Go through each setting and change its value
12. Check console and verify each setting has a corresponding event in console and the value displayed is the value that you selected
13. If the option changed is a switch then the event should be suffixed `_toggle` if it's a value change it should be suffixed with `_changed`

## Checklist
